snap/info_snap_yaml.go

```diff
@@ -292,6 +292,12 @@ func setAppsFromSnapYaml(y snapYaml, snap *Info) {
 
 func setHooksFromSnapYaml(y snapYaml, snap *Info) {
 	for hookName, yHook := range y.Hooks {
+		// Verify that the requested hook name is actually supported. If not,
+		// ignore it.
+		if !IsHookSupported(hookName) {
+			continue
```

Comment:
Indeed, as @niemeyer mentioned the idea is that hooks can be specified via assumes if they are required by the snap, in which case if they aren't present it'll cause an installation failure. Otherwise, the hooks are considered optional and if snapd doesn't support them they're simply ignored.